### ISSUE TITLE
Segfault in ucp_hello_world.c with big message sizes

### ISSUE THREAD

**Issue by vkocheganov** [openucx/ucx#977]
When change ucp_hello_world.c a little (increase size of test_str in particular) it fails:
client output:

```
[alexv@clx-orion-007 examples]$ ./a.out -n clx-orion-006
INFO: UCP_HELLO_WORLD mode = 0 server = clx-orion-006 port = 13337
UCX_NET_DEVICES=all
UCX_SHM_DEVICES=all
UCX_ACC_DEVICES=all
UCX_SELF_DEVICES=all
UCX_TLS=all
UCX_ALLOC_PRIO=md:sysv,md:posix,huge,md:*,mmap,heap
UCX_BCOPY_THRESH=0
UCX_RNDV_THRESH=auto
UCX_RNDV_THRESH_FALLBACK=200000
UCX_ZCOPY_THRESH=auto
UCX_BCOPY_BW=6081740800
UCX_LOG_DATA=0
UCX_MAX_WORKER_NAME=32
[0xf7fca740] local address length: 659
UCX address message was scheduled for send
[0xf7fca740] send handler called with status 0
UCX data message was received


----- UCP TEST SUCCESS ----



---------------------------

[1473774363.281840] [clx-orion-007:6932 :0]          rc_ep.c:426  UCX  WARN  destroying rc ep 0xc80228 with uncompleted operation 0x8eaf80
[1473774363.281848] [clx-orion-007:6932 :0]          rc_ep.c:426  UCX  WARN  destroying rc ep 0xc80228 with uncompleted operation 0x8eafa8
[0xf7fca740] receive handler called with status 0 (length 1000008)
[clx-orion-007:6932 :0] Caught signal 11 (Segmentation fault: address not mapped to object at address 0x2aaaae228090)
==== backtrace ====
 0 0x0000000000035670 killpg()  ??:0
 1 0x0000000000026578 __bswap_64()  /usr/include/bits/byteswap.h:111
 2 0x0000000000026578 uct_rc_mlx5_txqp_dptr_post()  /labhome/alexv/victor_folder/ucx/src/uct/ib/rc/accel/rc_mlx5_common.h:413
 3 0x0000000000026578 uct_rc_mlx5_txqp_bcopy_post()  /labhome/alexv/victor_folder/ucx/src/uct/ib/rc/accel/rc_mlx5_ep.c:27
 4 0x0000000000026578 uct_rc_mlx5_ep_am_bcopy()  /labhome/alexv/victor_folder/ucx/src/uct/ib/rc/accel/rc_mlx5_ep.c:245
 5 0x000000000000d20a uct_ep_am_bcopy()  /labhome/alexv/victor_folder/ucx/src/uct/api/uct.h:1252
 6 0x000000000000d20a ucp_proto_progress_am_bcopy_single()  /labhome/alexv/victor_folder/ucx/src/ucp/proto/proto_am.c:31
 7 0x000000000000ad38 ucp_request_try_send()  /labhome/alexv/victor_folder/ucx/src/ucp/core/ucp_request.c:111
 8 0x000000000000ad38 ucp_request_start_send()  /labhome/alexv/victor_folder/ucx/src/ucp/core/ucp_request.c:152
 9 0x000000000001ed06 uct_invoke_completion()  /labhome/alexv/victor_folder/ucx/src/uct/base/uct_iface.h:483
10 0x000000000001ed06 uct_rc_txqp_purge_outstanding()  /labhome/alexv/victor_folder/ucx/src/uct/ib/rc/base/rc_ep.c:434
11 0x000000000004aad6 ucs_class_call_cleanup_chain()  /labhome/alexv/victor_folder/ucx/src/ucs/type/class.c:50
12 0x0000000000022d68 uct_rc_verbs_ep_t_delete()  /labhome/alexv/victor_folder/ucx/src/uct/ib/rc/verbs/rc_verbs_ep.c:566
13 0x000000000000945a ucp_ep_destory_uct_eps()  /labhome/alexv/victor_folder/ucx/src/ucp/core/ucp_ep.c:214
14 0x000000000000945a ucp_disconnect_nb()  /labhome/alexv/victor_folder/ucx/src/ucp/core/ucp_ep.c:226
15 0x00000000000095be ucp_ep_destroy()  /labhome/alexv/victor_folder/ucx/src/ucp/core/ucp_ep.c:240
16 0x0000000000401d45 run_ucx_client()  ucp_hello_world.c:0
17 0x000000000040214f run_test()  ucp_hello_world.c:0
18 0x00000000004024ca main()  ??:0
19 0x0000000000021b15 __libc_start_main()  ??:0
20 0x0000000000401779 _start()  ??:0
===================
Segmentation fault
```

This is a diff with upstream ucp_hello_world.c version:

```
diff --git a/test/examples/ucp_hello_world.c b/test/examples/ucp_hello_world.c
index 8d976fe..c1ffbce 100644
--- a/test/examples/ucp_hello_world.c
+++ b/test/examples/ucp_hello_world.c
@@ -71,7 +71,7 @@ static ucp_address_t *peer_addr;
 static size_t local_addr_len;
 static size_t peer_addr_len;

-static char *test_str = "Hello UCP World!!!!";
+static char *test_str;// = "Hello UCP World!!!!";

 static int parse_cmd(int argc, char * const argv[], char **server_name);
 static int run_server();
@@ -325,6 +325,12 @@ static int run_ucx_server(ucp_worker_h ucp_worker)
     if (status != UCS_OK) {
         goto err;
     }
+    const int str_size = 1000000;
+    test_str = malloc(str_size*sizeof(*test_str));
+    int i;
+    for (i = 0; i < str_size-1; i++)
+      test_str[i] = 'h';
+    test_str[str_size-1] = '\0';

     msg_len = sizeof(*msg) + strlen(test_str) + 1;

     msg = calloc(1, msg_len);
@@ -335,6 +341,8 @@ static int run_ucx_server(ucp_worker_h ucp_worker)


     msg->data_len = msg_len - sizeof(*msg);

     snprintf((char *)msg->data, msg->data_len, "%s", test_str);
+    free(test_str);
+    test_str = NULL;

     request = ucp_tag_send_nb(client_ep, msg, msg_len,
                               ucp_dt_make_contig(1), tag,
```

**Comment by yosefe** [comment]
the endpoint is destroyed while it's still sending RNDV ATS message.
The fix is to flush before destroying. should be fixed by #969

**Comment by yosefe** [comment]
see also #981

**Comment by yosefe** [comment]
fixed by #981
